src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -334,6 +334,13 @@ export class VSCodeNotebookController implements Disposable, IVSCodeNotebookCont
             } called from ${new Error('').stack}`
         );
         this.isDisposed = true;
+        // Dispose all kernels associated with this controller
+        workspace.notebookDocuments.forEach((doc) => {
```

Comment:
Controller instances are not shared across notebooks.
We cannot dispose all other kernels associated with the same kernel.

Reply:
The latest [commit](https://github.com/microsoft/vscode-jupyter/pull/17097/commits/82a5ecd2d8926105ba7e76457b0638753ec711aa) filters to the `associatedDocuments`. Is that sufficient? If there's a better way, please let me know!